sszgen/main.go

```diff
@@ -418,13 +447,17 @@ func (e *env) print(order []string, experimental bool) (string, bool, error) {
 		if experimental {
 			getTree = e.getTree(name, obj)
 		}
-		objs = append(objs, &Obj{
+		o := &Obj{
 			HashTreeRoot: e.hashTreeRoot(name, obj),
 			GetTree:      getTree,
 			Marshal:      e.marshal(name, obj),
 			Unmarshal:    e.unmarshal(name, obj),
 			Size:         e.size(name, obj),
-		})
+		}
+		if len(obj.opts) == 1 && obj.opts[0] == "no-htr" {
+			o.HashTreeRoot = ""
+		}
```

Comment:
What about:
```
	o := &Obj{
			GetTree:      getTree,
			Marshal:      e.marshal(name, obj),
			Unmarshal:    e.unmarshal(name, obj),
			Size:         e.size(name, obj),
		})
		if len(obj.opts) == 1 && obj.opts[0] == "no-htr" {
			o.HashTreeRoot = ""
		} else {
		        o.HashTreeRoot = e.hashTreeRoot(name, obj)
		}
```

My reasoning is that this would enable the flag to be used to prevent `hashTreeRoot` from running when there is some bug in the way that the codegen handles a particular structure. Also, if we're going to discard the output anyway, might as well not run it.

Reply:
Yes, I 100% agree. I did it the odd way because I had trouble with this approach, which made me assume that the order of method generation is relevant. But it works fine now :-)